Source/JavaScriptCore/runtime/ExceptionHelpers.cpp

```diff
@@ -62,8 +62,13 @@ String errorDescriptionForValue(JSGlobalObject* globalObject, JSValue v)
         return tryMakeString('"', string, '"');
     }
 
-    if (v.isSymbol())
-        return asSymbol(v)->descriptiveString();
+    if (v.isSymbol()) {
+        auto expectedDescription = asSymbol(v)->tryGetDescriptiveString();
+        if (expectedDescription)
+            return expectedDescription.value();
+        ASSERT(expectedDescription.error() == ErrorTypeWithExtension::OutOfMemoryError);
+        return String("Symbol"_s);
```

Comment:
We should throw an OOM exception here.

Reply:
Why?  Is it not a better developer experience to know about the original error rather than the unrelated OOME?  For example, in the attached repro test case, they would get ""TypeError: Symbol is not a constructor (evaluating 'new s(2)')", which tells them more clearly what the issue is.